### ISSUE TITLE
HTML5 open web platform

### ISSUE THREAD

**Issue by Gizzz** [yangshun/front-end-interview-handbook#138]
In HTML section there is a question: 
> Consider HTML5 as an open web platform. What are the building blocks of HTML5?  

And the answer is from MDN's HTML5 docs:
[https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/HTML5)

Probably something different is meant in the original question. I understand it as a question about tech stack of open web platform. Please take a loot at the diagrams on this pages to see what I mean: 
[http://tess.oconnor.cx/2009/05/what-the-web-platform-is](http://tess.oconnor.cx/2009/05/what-the-web-platform-is)
[https://hsivonen.fi/web-stack/](https://hsivonen.fi/web-stack/)

**Comment by yangshun** [comment]
I agree. A better answer is welcome. Off the top of my head, and based on the diagram:

![henri-stack-v2](https://user-images.githubusercontent.com/1315101/73194252-dd179000-4166-11ea-8c79-a0e7b2276569.png)

I think some components are: HTML, JavaScript, CSS, HTTP, Security, Performance, Accessibility, Web Media (images, audio, video), WebGL, Storage (Cookies, Local/Session Storage, IndexedDB), Service Workers, Performance, Animation, Web Asssembly, Search Engine Optimization, and more. Would be nice to have some further elaboration on each component.

**Comment by captainWaheed** [comment]
This diagram covers mostly all of the possible blocks.
I just saw this: on [https://flowerszhong.github.io/2013/11/20/html-questions.html](url)

![Sementics](https://user-images.githubusercontent.com/86144678/179366244-ae2ee19e-a375-44db-b540-d7ead12c416e.jpg)

**Comment by ishratUmar18** [comment]
### HTML5 building blocks:

1. Semantic elements such as the header, main,  section, footer, etc.
2.  Introduced new attributes for the form's <input> tag.
3. The local storage of data that allows webpages to function more efficiently offline.
4. Embedding multimedia is possible without third-party plugins.
5. Canvas that offers a much greater range of presentation options.
6. Document Object Model (DOM) and APIs.
7. Support for mathematical formulas and SVG files.

[_Reference_
](https://www.tutorialspoint.com/html5/html5_overview.htm)

**Comment by sdkdeepa** [comment]
This is the latest  I could find on this. https://html.spec.whatwg.org/
